modules/kubernetes/logs/scrubs/json-nulls.river

```diff
@@ -0,0 +1,32 @@
+/*
+Module: scrub-json-nulls
+Description: Removes any json properties with a null value
+Arguments:
+  forward_to: Module to forward the output to
+*/
+argument "forward_to" {
+  optional = false
+}
+
+export "process" {
+  value = loki.process.scrub_json_nulls
+}
+
+loki.process "scrub_json_nulls" {
+  forward_to = [argument.forward_to.value]
+
+  // check logs.agent.grafana.com/scrub-nulls annotation, if true remove any json property whose value is set to null
+  // this can reduce the overall # of bytes sent and stored in Loki
+  stage.match {
+    selector = "{logs_agent_grafana_com_scrub_nulls=~\"(?i)(dotnet-?json|istio|(generic-?)?json|log4j-?json|(otel|open-?telemetry)(-?json)?|python-?json)\",logs_agent_grafana_com_scrub_nulls=~\"(?i)true\"}"
+    pipeline_name = "pipeline for annotation || logs.agent.grafana.com/scrub-null: true"
+
+    // remove null properties
+    stage.replace {
+      // unescaped regex: (\s*,\s*("[^"]+"\s*:\s*null)|("[^"]+"\s*:\s*null)\s*,\s*)
```

Comment:
This is a great example (and all the other regexs) why we need strings that dont require escaping

Reply:
Agreed